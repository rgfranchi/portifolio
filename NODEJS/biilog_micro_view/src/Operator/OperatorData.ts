/**
 * Ações a serem executadas com servidorer para manutenção da entidade.
 * Matem premissas de dadados ex: validação, valor vazio, etc...
 */
import {
  createUpdate,
  dropById,
  listAll,
  searchById,
} from "../helper/CRUDDataHelper";

import {
  IDataDelete,
  IDataList,
  IDataSave,
  IDataValue,
} from "../interfaces/DataInterface";

/**
 * Interface de controle dos objetos da Entidade.
 */
export interface IOperator {
  id: number;
  nome: string;
  codOperador: string;
  senha: string;
  rfid: string;
  digital: string;
  sexo: string; // ENUM
  funcao: string;
  acessoDataInicial: string;
  acessoDataFinal: string;
  email: string;
  observacao: string;
  companyId: number;
}

/**
 * Valores padrão vazio para preenchimento da entidade
 */
export const emptyOperator: IOperator = {
  id: 0,
  nome: "",
  codOperador: "",
  senha: "",
  rfid: "",
  digital: "",
  sexo: "",
  funcao: "",
  acessoDataInicial: "", // data definidas no form para possibilitar testes.
  acessoDataFinal: "", // data definidas no form para possibilitar testes.
  email: "",
  observacao: "",
  companyId: 0,
};

/**
 * URL principal de acesso a informação da entidade.
 */
const pathURL = "/operator";

/**
 * Texto das entidades no singular e plural.
 */
export const displayName = { many: "Operadores", one: "Operador" };

export const getListByCompany = async (
  companyId: string
): Promise<IDataList<IOperator>> => {
  return await listAll<IOperator>({
    displayName: displayName,
    pathURL: pathURL,
    pathURI: `/findByCompany/${companyId}`,
  });
};

export const getList = async (): Promise<IDataList<IOperator>> => {
  return await listAll<IOperator>({
    displayName: displayName,
    pathURL: pathURL,
  });
};

export const findById = async (id: string): Promise<IDataValue<IOperator>> => {
  return await searchById<IOperator>({
    displayName: displayName,
    pathURL: pathURL,
    id: id,
  });
};

export const saveData = async (
  value: IOperator
): Promise<IDataSave<IOperator>> => {
  return await createUpdate<IOperator>({
    displayName: displayName,
    pathURL: pathURL,
    valueId: value.id.toString(),
    value: value,
    textShow: value.nome,
  });
};

export const deleteData = async (id: string): Promise<IDataDelete> => {
  return await dropById({
    displayName: displayName,
    pathURL: pathURL,
    id: id,
  });
};
