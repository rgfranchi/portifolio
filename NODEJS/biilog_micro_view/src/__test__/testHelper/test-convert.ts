export const convertIdToNumeric = (data: any) => {
  if (Array.isArray(data)) {
    data.forEach((element) => {
      element = convertIdToNumeric(element);
    });
  }

  if (typeof data === "object" && data) {
    Object.entries(data).forEach((value, key) => {
      const k = value[0];
      const v = value[1];
      if (typeof v === "object") {
        data[k] = convertIdToNumeric(v);
      }
      if (k === "id") {
        const parse = parseInt(data[k]);
        if (!isNaN(parse)) {
          data[k] = parse;
        }
      }
    });
  }
  return data;
};

export const convertDateYYYYMMDDToDD_MM_YYYY = (dateSrt: string) => {
  const dArr = dateSrt.split("-"); // ex input "2010-01-18"
  return dArr[2] + "/" + dArr[1] + "/" + dArr[0]; //ex out: "18/01/2010"
};
