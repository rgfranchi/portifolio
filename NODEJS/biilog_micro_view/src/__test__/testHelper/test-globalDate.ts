/**
 * Fixa a data e hora do sistema para teste.
 * Realizar chamada no beforeAll(() => {});
 * Criar variÃ¡vel global : const realDate = Date.bind(global.Date);
 * E incluir no afterAll(() => {global.Date = realDate;});
 * @param YYYY
 * @param MM
 * @param DD
 * @param HH
 * @param mm
 * @param ss
 */

export const fixedGlobalDate = (
  YYYY: number,
  MM: number,
  DD: number,
  HH: number,
  mm: number,
  ss: number
) => {
  // https://github.com/facebook/jest/issues/2234#issuecomment-573621597
  const fixedDate = new Date(YYYY, MM, DD, HH, mm, ss);
  const d = Date;
  // This will only mock any Date objects instantiated with new
  // and not Date.now().
  const mockGlobal = global;
  // @ts-ignore
  mockGlobal.Date = jest.fn(() => fixedDate);
  mockGlobal.Date.parse = d.parse;
  mockGlobal.Date.UTC = d.UTC;
  mockGlobal.Date.now = d.now;
};
