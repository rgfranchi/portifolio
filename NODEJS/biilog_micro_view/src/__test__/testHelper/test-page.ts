import toJson from "enzyme-to-json";

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} warpper -> componente que será testado, verifica Snapshot
 */
export const verifyPageContents = (values: any, wrapper: any) => {
  values.find.forEach((search: any) => {
    // console.log(search);
    expect(wrapper.find(search).exists()).toBeTruthy();
  });
  // console.log(toJson(wrapper));
  expect(toJson(wrapper)).toMatchSnapshot();
};

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} warpper -> componente que será testado, verifica Snapshot
 */
export const verifyNotAccess = (values: any, wrapper: any) => {
  // console.log(values.find);
  values.find.forEach((search: any) => {
    // console.log(search);
    expect(wrapper.find(search).exists()).toBeFalsy();
  });
};

/**
 * verifica o conteúdo de construção dos componentes Page.
 * @param {*} values -> array defaultPageText de test-text-defaults.js
 * @param {*} history -> histórico da url para validação.
 */
export const loadHistoryPageContents = (values: any, history: any) => {
  history.entries[0].key = values.historyKey1;
  if (values.url1 !== "") {
    history.push(values.url1);
    history.entries[1].key = values.historyKey2;
  }
};
