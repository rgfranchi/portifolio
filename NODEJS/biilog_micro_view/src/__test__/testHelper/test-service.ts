import {
  screen,
  waitForElement,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { confirmModal } from "../../redux/modalSlice";

/**
 * Verifica se os elementos de save da tela foram executados com sucesso.
 * @param {boolean} isSuccess
 * @param {*} store
 */
export const verifyScreenAfterSave = async (isSuccess: boolean, store: any) => {
  // verifica se o modal Loading está ativo
  expect(screen.getByText(/Loading/i)).toBeInTheDocument();
  // verifica mensagem de falha.
  expect(store.getState().alert.hidden).toBe(true);
  // aguarda modal finalizar
  await waitForElementToBeRemoved(() => screen.getByText(/Loading/i));

  // Verifica estado dos elementos.
  if (isSuccess) {
    expect(store.getState().modals.success.show).toBe(true);
    expect(store.getState().modals.success.textButton).toMatch(/Retornar/i);
  } else {
    expect(store.getState().alert.hidden).toBe(false);
    expect(store.getState().alert.message).toMatch(/ERROR STATUS\[/i);
  }
};

/**
 * Executa comando de exclusão da tela List
 * @param {number} buttonPosition
 * @param {*} store
 */
export const verifyExecuteScreenDelete = async (
  buttonPosition: number,
  store: any
) => {
  // clica sobre o segundo link
  userEvent.click(screen.getAllByRole("button")[buttonPosition]);
  // verifica se alterou para exibir modal
  expect(store.getState().modals.confirm.show).toBe(true);
  // variável de NÃO executado.
  expect(store.getState().modals.confirm.executed).toBe(0);
  // realiza chamada de execução do modal.
  store.dispatch(confirmModal({ show: true, executed: 1 }));
  expect(store.getState().modals.confirm.executed).toBe(1);

  await waitForElement(() => screen.getByText(/Carregando/i));

  await waitForElementToBeRemoved(() => screen.getByText(/Carregando/i));
};

/**
 * Verifica e guarda carregar valores no tela, Create e Update
 * @param {boolean} isSuccess
 * @param {"modal" | "alert"} type "modal" | "alert"
 * @param {*} store
 */
export const loadingScreenValues = async (
  isSuccess: boolean,
  type: "modal" | "alert",
  store: any
) => {
  // state do loading exibido
  expect(store.getState().modals.loading.show).toBe(true);
  // elementos carregando Página de Loading
  expect(screen.getByText(/Carregando/i)).toBeInTheDocument();
  // verifica mensagem de falha.
  expect(store.getState().alert.hidden).toBe(true);
  // aguarda loading finalizar.
  await waitForElementToBeRemoved(() => screen.getByText(/Carregando/i));
  if (type === "modal") {
    if (isSuccess) {
      // state do loading oculto
      expect(store.getState().modals.loading.show).toBe(false);
    } else {
      expect(store.getState().modals.fail.show).toBe(true);
      expect(store.getState().modals.fail.textFail).toMatch(/vazia/i);
    }
  } else {
    expect(store.getState().alert.hidden).toBe(false);
    expect(store.getState().alert.message).toMatch(/vazia/i);
  }
};
