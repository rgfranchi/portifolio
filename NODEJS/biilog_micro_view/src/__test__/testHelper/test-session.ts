import { listGroupMock } from "../server/serverGroup";
import {
  ISessionAccess,
  ISessionSlice,
} from "../../interfaces/SessionInterface";

/**
 * Todo: Utilizar interface :ISessionSlice
 */
export const objMockEmptySession: ISessionSlice = {
  userId: 0,
  company: { id: 0, nome: "" },
  group: { id: 0, nome: "", access: [] },
};

export const objMockSession: ISessionSlice = {
  userId: 3,
  company: { id: 2, nome: "WINE SESSION" },
  group: listGroupMock[0],
};

export const objMockMenuAccess: ISessionAccess[] = [
  {
    path: "/company",
    page: {
      create: {
        label: "Nova Empresa",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/operator",
    page: {
      create: {
        label: "Inserir Operador",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/user",
    page: {
      create: {
        label: "Novo Usuário",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/machine",
    page: {
      create: {
        label: "Nova Máquina",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
  {
    path: "/forklift",
    page: {
      create: {
        label: "Inserir Equipamento",
        path: "/new",
      },
      delete: {
        path: "/delete",
      },
      update: {
        path: "/update",
      },
    },
  },
];
