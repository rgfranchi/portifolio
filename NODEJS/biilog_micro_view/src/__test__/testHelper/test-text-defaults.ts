import { IAlertSlice, ICompanySlice } from "../../interfaces/ReduxInterface";

export let defaultMessageTextData = {
  vaziaResponse: /vazia/i,
  falhaAcessarServidor: /Falha ao acessar servidor/i,
  idNaoDefinido: /ID não definido/i,
  criadoValor: "VALOR CRIADO XPTO",
  criadoSucesso: /Criado com sucesso/i,
  atualizadoValor: "VALOR ATUALIZADO XPTO",
  atualizadoSucesso: /Atualizado com sucesso/i,
  excluidoSucesso: "DELETED",
  erro208: "RETORNA_ERRO_208", // resposta Valida porem não esperada.
  erro208Mensagem: /INVALIDO Status/i,
  erro400: "RETORNA_ERRO_400", // resposta Valida porem não esperada.
  erro400Mensagem: /ERROR STATUS/i,
};

export let defaultFormTextValidate = {
  preenchimento: /Preenchimento obrigatório/i,
  minimo3caracteres: /com no mínimo 3 caracteres/i,
  emailInvalido: /e-mail inválido/i,
  cepFormato: /formato de 99999-999/i,
  cnpjInvalido: /CNPJ inválido/i,
  selecioneOpcao: /Selecione Uma Opção/i,
};

export let defaultPageText = {
  list: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_LIST1",
    url1: "",
    historyKey2: "",
    find: ["TitleComponent", "ButtonCreate", "List"],
  },
  create: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_NEW1",
    url1: "//new",
    historyKey2: "KEY_NEW2",
    find: ["TitleComponent", "CreateForm"],
  },
  update: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_UPDATE1",
    url1: "//update/2",
    historyKey2: "KEY_UPDATE2",
    find: ["TitleComponent", "UpdateForm"],
  },
};

export let alertSliceDefault: IAlertSlice = {
  hidden: true,
  message: null,
  variant: "primary",
};

export let companySliceDefault: ICompanySlice = {
  id: 3,
  nome: "Company Slice Mock",
};

// @ts-ignore
export let modalSliceDefault: IConfirmModal = {
  loading: {
    show: false,
    message: null,
  },
  confirm: {
    show: false,
    executed: 0, // incrementa a cada nova tentativa.
    disabledButtons: false,
    textTitle: null,
    textBody: null,
    textAction: "Executar",
    errorExecute: null,
    variantAction: "primary",
  },
  fail: {
    show: false,
    redirect: null,
    textFail: "Falha ao Executar",
    textButton: "Falhou ... ",
  },
  success: {
    show: false,
    redirect: null,
    textSuccess: "Sucesso!!!",
    textButton: "OK",
  },
};
