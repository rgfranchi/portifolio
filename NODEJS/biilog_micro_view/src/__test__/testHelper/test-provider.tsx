import React from "react";
import { MemoryRouter, Route, Router } from "react-router-dom";

import { objMockSession } from "./test-session";

import { Provider } from "react-redux";

//componentes de tela.
import { ModalLoading } from "../../components/ModalsComponent";

/**
 * Renderiza valores para teste.
 * Utilizando o render do framework @testing-library/react
 * Componente ModalLoading.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @returns
 */
export const renderProviderRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  // @ts-ignore
  function Wrapper({ children }) {
    return providerRouter(children, history, store, modal);
  }
  // @ts-ignore
  return render(ui, { wrapper: Wrapper });
};

/**
 * ConstrÃ³i com base no renderProviderRouter incluindo MemoryRouter e Router.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @param {*} initialEntries
 * @param {*} path
 * @returns
 */
export const renderProviderMemoryRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  initialEntries: any,
  path: string,
  modal: boolean = true
) => {
  const wrapper = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Route path={path}>{ui}</Route>
    </MemoryRouter>
  );
  return renderProviderRouter(render, wrapper, history, store, modal);
};

/**
 * Cria as funcionalidades do Provider e Router
 * Utilizando o mount do framework Enzyme
 * @param {*} mount
 * @param {*} children
 * @param {*} history
 * @param {*} store
 * @param {boolean} modal
 * @returns
 */
export const mountProviderRouter = async (
  mount: any,
  children: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  return await mount(providerRouter(children, history, store, modal));
};

export const createSessionCurrentUser = (store: any) => {
  store.dispatch({
    type: "session/currentUser",
    payload: objMockSession,
  });
};

const providerRouter = (
  children: any,
  history: any,
  store: any,
  modal: boolean
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        {modal && <ModalLoading />}
        {children}
      </Router>
    </Provider>
  );
};
