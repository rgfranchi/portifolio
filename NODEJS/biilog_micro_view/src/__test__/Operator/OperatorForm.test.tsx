import React from "react";

import { act } from "react-dom/test-utils";

import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { createMemoryHistory } from "history";

import store from "../../redux/store";

import { renderProviderRouter } from "../testHelper/test-provider";

import { defaultFormTextValidate } from "../testHelper/test-text-defaults";

import {
  operatorMock,
  loadOperatorValidValues,
  loadOperatorInvalidValues,
} from "../server/serverOperator";

import OperatorForm from "../../Operator/OperatorForm";
import { emptyOperator } from "../../Operator/OperatorData";

let wrapper: any = null;
let history: any = null;
const objectMock = operatorMock;
const objectEmpty = {
  ...emptyOperator,
  ...{
    acessoDataFinal: "2021-01-08",
    acessoDataInicial: "2021-01-08",
  },
};
const submitMock = jest.fn();

beforeEach(() => {
  cleanup();
  jest.setTimeout(20000);
  history = createMemoryHistory();
});

describe("OperatorForm snapshot", () => {
  test("empty", async () => {
    history.entries[0].key = "EMPTY_01";
    await setWrapperDefault(objectEmpty, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
  test("with data", async () => {
    history.entries[0].key = "DATA_01";
    await setWrapperDefault(objectMock, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
});

describe("OperatorForm validade", () => {
  test("valid", async () => {
    await setWrapperDefault(objectEmpty, submitMock);
    await act(async () => {
      await loadOperatorValidValues(userEvent, screen);
    });
    expect(
      screen.queryByText(defaultFormTextValidate.preenchimento)
    ).toBeNull();
    expect(
      screen.queryByText(defaultFormTextValidate.minimo3caracteres)
    ).toBeNull();
    expect(
      screen.queryByText(defaultFormTextValidate.selecioneOpcao)
    ).toBeNull();
    expect(
      screen.queryByText(defaultFormTextValidate.emailInvalido)
    ).toBeNull();
    expect(
      screen.queryByText(/Acesso Inicial deve ser ANTERIOR a Final/i)
    ).toBeNull();
    expect(
      screen.queryByText(/Acesso Final deve ser POSTERIOR a Inicial/i)
    ).toBeNull();
    await act(async () => {
      fireEvent.click(screen.getByText("Salvar"));
    });
    expect(submitMock).toHaveBeenCalledTimes(1);
  });

  test("invalid", async () => {
    await act(async () => {
      await setWrapperDefault(objectEmpty, submitMock);
      fireEvent.click(screen.getByText("Salvar"));
    });

    expect(
      screen.queryAllByText(defaultFormTextValidate.preenchimento).length
    ).toBe(4);

    await act(async () => {
      await loadOperatorInvalidValues(userEvent, screen);
    });

    expect(
      screen.queryAllByText(defaultFormTextValidate.preenchimento).length
    ).toBe(2);
    expect(
      screen.queryAllByText(defaultFormTextValidate.minimo3caracteres).length
    ).toBe(1);
    expect(
      screen.queryAllByText(defaultFormTextValidate.selecioneOpcao).length
    ).toBe(1);
    expect(
      screen.queryAllByText(defaultFormTextValidate.emailInvalido).length
    ).toBe(1);
    expect(
      screen.queryAllByText(/Acesso Inicial deve ser ANTERIOR a Final/i).length
    ).toBe(1);
    expect(
      screen.queryAllByText(/Acesso Final deve ser POSTERIOR a Inicial/i).length
    ).toBe(1);
  });
});

const setWrapperDefault = async (object: any, handleSubmit: any = () => {}) => {
  await act(async () => {
    wrapper = await renderProviderRouter(
      render,
      <OperatorForm initValues={object} submitForm={handleSubmit} />,
      history,
      store
    );
  });
};

// Enzime.configure({ adapter: new Adapter() });

// beforeEach(() => {
//   cleanup();
// });

// test("OperatorForm snapshot empty", () => {
//   const empty = {
//     ...emptyOperator,
//     ...{
//       acessoDataFinal: "2021-01-08",
//       acessoDataInicial: "2021-01-08",
//     },
//   };
//   const wrapper = mount(
//     <OperatorForm
//       initValues={empty}
//       submitForm={() => {
//         console.log("Submit");
//       }}
//     />
//   );
//   // valores vazio
//   expect(toJson(wrapper)).toMatchSnapshot();
// });

// test("OperatorForm snapshot valores", () => {
//   const wrapper = mount(
//     <OperatorForm
//       initValues={listOperatorMock[1]}
//       submitForm={() => {
//         console.log("Submit");
//       }}
//     />
//   );
//   expect(toJson(wrapper)).toMatchSnapshot();
// });

// test("OperatorForm validos Create", async () => {
//   jest.setTimeout(20000);
//   const submitMock = jest.fn();
//   render(<OperatorForm initValues={emptyOperator} submitForm={submitMock} />);
//   await act(async () => {
//     userEvent.selectOptions(screen.getByLabelText("Sexo"), ["Masculino"]);
//     await userEvent.type(
//       screen.getByLabelText("E-mail"),
//       "operador@teste.com",
//       {
//         delay: 100,
//       }
//     );
//     await userEvent.type(screen.getByLabelText("Nome"), "NOME TESTE", {
//       delay: 100,
//     });
//     await userEvent.type(screen.getByLabelText("Código"), "Código 9999", {
//       delay: 100,
//     });
//     await userEvent.type(screen.getByLabelText("Tipo"), "Operador Teste", {
//       delay: 100,
//     });
//     await userEvent.type(screen.getByLabelText("Código RFID"), "RFID_TESTE", {
//       delay: 100,
//     });
//     await userEvent.type(screen.getByLabelText("Senha"), "SENHA_TESTE", {
//       delay: 100,
//     });
//     await userEvent.type(
//       screen.getByLabelText("Observações"),
//       "Operador Observações de texto",
//       {
//         delay: 100,
//       }
//     );
//   });
//   // verifica se não possui msg de erro.
//   expect(screen.queryByText("Preenchimeto obrigatório")).toBeNull();
//   expect(screen.queryByText("Nome com no mínimo 3 caracteres")).toBeNull();
//   expect(screen.queryByText("Selecione Uma Opção")).toBeNull();
//   expect(
//     screen.queryByText("Acesso Inicial deve ser ANTERIOR a Final")
//   ).toBeNull();
//   expect(
//     screen.queryByText("Acesso Final deve ser POSTERIOR a Inicial")
//   ).toBeNull();

//   // expect(screen.queryByText("Endereço de e-mail inválido")).toBeNull();
//   expect(screen.queryByText("Selecione uma empresa .... ")).toBeNull();
//   // submit form.
//   await act(async () => {
//     fireEvent.click(screen.getByText("Salvar"));
//   });
//   expect(submitMock).toHaveBeenCalledTimes(1);

//   // recebe data atual apara comparar.
//   const nowString = dateTimeNowToString(false);
//   expect(submitMock.mock.calls[0][0]).toEqual({
//     acessoDataFinal: nowString,
//     acessoDataInicial: nowString,
//     codOperador: "Código 9999",
//     companyId: 0, // está no contexto da aplicação.
//     email: "operador@teste.com",
//     id: 0,
//     nome: "NOME TESTE",
//     observacao: "Operador Observações de texto",
//     rfid: "RFID_TESTE",
//     senha: "SENHA_TESTE",
//     sexo: "Masculino",
//     tipo: "Operador Teste",
//   });
// });

// test("OperatorForm validade inválidos", async () => {
//   jest.setTimeout(20000);
//   const submitMock = jest.fn();
//   render(<OperatorForm initValues={emptyOperator} submitForm={submitMock} />);
//   // executa post sem valor preenchido
//   await act(async () => {
//     fireEvent.click(screen.getByText("Salvar"));
//     // await new Promise((r) => setTimeout(r, 500)); // aguarda resposta do banco de dados
//   });
//   // verifica os campos em branco Nome / Código / Tipo / E-mail
//   expect(screen.queryAllByText("Preenchimento obrigatório").length).toBe(4);
//   expect(screen.queryByText("Selecione Uma Opção")).toBeTruthy();
//   await act(async () => {
//     await userEvent.type(screen.getByLabelText("Nome"), "AB", {
//       delay: 300,
//     });
//     await userEvent.type(
//       screen.getByLabelText("Acesso Inicial"),
//       "01/01/2020",
//       {
//         delay: 300,
//       }
//     );
//     await userEvent.type(screen.getByLabelText("Acesso Final"), "01/01/2000", {
//       delay: 300,
//     });
//     await userEvent.type(screen.getByLabelText("E-mail"), "operador@xxxxx", {
//       delay: 300,
//     });
//   });
//   // verifica os campo Nome Com menos de 3 caracteres
//   expect(screen.queryByText("Nome com no mínimo 3 caracteres")).toBeTruthy();
//   // verifica se data inicial é menor que final
//   expect(
//     screen.queryByText("Acesso Inicial deve ser ANTERIOR a Final")
//   ).toBeTruthy();
//   expect(
//     screen.queryByText("Acesso Final deve ser POSTERIOR a Inicial")
//   ).toBeTruthy();
//   // verifica e-mail incorreto
//   expect(screen.queryByText("Endereço de e-mail inválido")).toBeTruthy();
//   // verifica id da empresa
// });
