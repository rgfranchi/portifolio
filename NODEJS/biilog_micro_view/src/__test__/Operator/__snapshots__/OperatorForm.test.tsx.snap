// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorForm snapshot empty 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-7"
      >
        <label
          class="form-label"
          for="ctrlNome"
        >
          Nome
        </label>
        <input
          class="form-control"
          id="ctrlNome"
          name="nome"
          placeholder="Nome do Operador"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col"
      >
        <label
          class="form-label"
          for="ctrlCodOperador"
        >
          Código
        </label>
        <input
          class="form-control"
          id="ctrlCodOperador"
          name="codOperador"
          placeholder="Código do operador"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlSenha"
        >
          Senha
        </label>
        <input
          class="form-control"
          id="ctrlSenha"
          name="senha"
          placeholder="Senha Operador"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlRFID"
        >
          Código RFID
        </label>
        <input
          class="form-control"
          id="ctrlRFID"
          name="rfid"
          placeholder="Código RFID"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlFuncao"
        >
          Função
        </label>
        <input
          class="form-control"
          id="ctrlFuncao"
          name="funcao"
          placeholder="Função Operador"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlAcessoDataInicial"
        >
          Acesso Inicial
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              id="ctrlAcessoDataInicial"
              name="acessoDataInicial"
              type="text"
              value="07/01/2021"
            />
          </div>
        </div>
        <div
          class="invalid-feedback"
          style="display: block;"
        />
      </div>
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlAcessoDataFinal"
        >
          Acesso Final
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              id="ctrlAcessoDataFinal"
              name="acessoDataFinal"
              type="text"
              value="07/01/2021"
            />
          </div>
        </div>
        <div
          class="invalid-feedback"
          style="display: block;"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlEMAIL"
        >
          E-mail
        </label>
        <input
          class="form-control"
          id="ctrlEMAIL"
          name="email"
          placeholder="email@email.com"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-2"
      >
        <label
          class="form-label"
          for="ctrlSexo"
        >
          Sexo
        </label>
        <select
          class="form-control"
          id="ctrlSexo"
          name="sexo"
        >
          <option
            value=""
          >
            Selecione ... 
          </option>
          <option
            value="Masculino"
          >
            Masculino
          </option>
          <option
            value="Feminino"
          >
            Feminino
          </option>
        </select>
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="ctrlObservacao"
      >
        Observações
      </label>
      <textarea
        class="form-control"
        id="ctrlObservacao"
        name="observacao"
        placeholder="Informações sobre o operador .... "
        rows="4"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`OperatorForm snapshot with data 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-7"
      >
        <label
          class="form-label"
          for="ctrlNome"
        >
          Nome
        </label>
        <input
          class="form-control"
          id="ctrlNome"
          name="nome"
          placeholder="Nome do Operador"
          type="text"
          value="NOME TESTE"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col"
      >
        <label
          class="form-label"
          for="ctrlCodOperador"
        >
          Código
        </label>
        <input
          class="form-control"
          id="ctrlCodOperador"
          name="codOperador"
          placeholder="Código do operador"
          type="text"
          value="88888"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlSenha"
        >
          Senha
        </label>
        <input
          class="form-control"
          id="ctrlSenha"
          name="senha"
          placeholder="Senha Operador"
          value="Senha 9999"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlRFID"
        >
          Código RFID
        </label>
        <input
          class="form-control"
          id="ctrlRFID"
          name="rfid"
          placeholder="Código RFID"
          type="text"
          value="RF 8888"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlFuncao"
        >
          Função
        </label>
        <input
          class="form-control"
          id="ctrlFuncao"
          name="funcao"
          placeholder="Função Operador"
          type="text"
          value="OPERADOR"
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlAcessoDataInicial"
        >
          Acesso Inicial
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              id="ctrlAcessoDataInicial"
              name="acessoDataInicial"
              type="text"
              value="10/01/2021"
            />
          </div>
        </div>
        <div
          class="invalid-feedback"
          style="display: block;"
        />
      </div>
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlAcessoDataFinal"
        >
          Acesso Final
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              id="ctrlAcessoDataFinal"
              name="acessoDataFinal"
              type="text"
              value="10/01/2021"
            />
          </div>
        </div>
        <div
          class="invalid-feedback"
          style="display: block;"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlEMAIL"
        >
          E-mail
        </label>
        <input
          class="form-control"
          id="ctrlEMAIL"
          name="email"
          placeholder="email@email.com"
          value="operador@teste.com"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-2"
      >
        <label
          class="form-label"
          for="ctrlSexo"
        >
          Sexo
        </label>
        <select
          class="form-control"
          id="ctrlSexo"
          name="sexo"
        >
          <option
            value=""
          >
            Selecione ... 
          </option>
          <option
            value="Masculino"
          >
            Masculino
          </option>
          <option
            value="Feminino"
          >
            Feminino
          </option>
        </select>
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="ctrlObservacao"
      >
        Observações
      </label>
      <textarea
        class="form-control"
        id="ctrlObservacao"
        name="observacao"
        placeholder="Informações sobre o operador .... "
        rows="4"
      >
        Isto é um teste teste
      </textarea>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;
